src/Network/Monitoring/Riemann/Json.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE DeriveAnyClass    #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# OPTIONS_GHC  -fno-warn-orphans #-}
+module Network.Monitoring.Riemann.Json where
+
+import           Data.Aeson                                 (FromJSON, ToJSON,
+                                                             Value (String),
+                                                             parseJSON, toJSON,
+                                                             withObject, (.!=),
+                                                             (.:), (.:?))
+import           Data.Aeson.Types                           (typeMismatch)
+import qualified Data.Text                                  as Text
+import           Network.Monitoring.Riemann.Proto.Attribute as PA
+import           Network.Monitoring.Riemann.Proto.Event     as PE
+import qualified Text.ProtocolBuffers.Header                as P'
+
+instance ToJSON P'.Utf8 where
+  toJSON v = String (Text.pack (P'.uToString v))
+
+instance FromJSON P'.Utf8 where
+  parseJSON (String s) = pure. P'.uFromString . Text.unpack $ s
```

Comment:
You can use `withText` here http://hackage.haskell.org/package/aeson-1.4.0.0/docs/Data-Aeson-Types.html#v:withText

```
parseJSON = withText $ (P'.uFromString . Text.unpack)
```

Should do it

Reply:
Done